qa/tasks/cephfs/test_mds_metrics.py

```diff
@@ -292,14 +292,29 @@ def test_query_mds_filter(self):
         log.debug("metrics={0}".format(metrics))
         self.assertTrue(valid)
 
+        filtered_mds = 1
+        def verify_filtered_mds_rank_metrics(metrics):
+        # checks if the metrics has only client_metadata and
+        # global_metrics filtered using --mds_rank=1
+            global_metrics = metrics['global_metrics']
+            client_metadata = metrics['client_metadata']
+            mds_metrics = metrics['metrics']
+            if len(mds_metrics) != 2 or f"mds.{filtered_mds}" not in mds_metrics:
+                return False
+            if len(global_metrics) != TestMDSMetrics.CLIENTS_REQUIRED or\
+                    len(client_metadata) != TestMDSMetrics.CLIENTS_REQUIRED:
+                return False
+            if len(set(global_metrics) - set(mds_metrics[f"mds.{filtered_mds}"])) or\
+                    len(set(client_metadata) - set(mds_metrics[f"mds.{filtered_mds}"])):
+                return False
+            return True
         # initiate a new query with `--mds_rank` filter and validate if
         # we get metrics *only* from that mds.
-        filtered_mds = 1
-        valid, metrics = self._get_metrics(
-            self.verify_mds_metrics(client_count=TestMDSMetrics.CLIENTS_REQUIRED,
-                                    ranks=[filtered_mds]), 30, '--mds_rank={}'.format(filtered_mds))
+        valid, metrics = self._get_metrics(verify_filtered_mds_rank_metrics, 30,
+                                           '--mds_rank={}'.format(filtered_mds))
         log.debug("metrics={0}".format(metrics))
-        self.assertTrue(valid)
+        self.assertTrue(valid, "Incorrect 'ceph fs perf stats' output"
+                        f"with filtered '--mds_rank={filtered_mds}'")
```

Comment:
Did you mean s/filtered/filter ?
nit: put f at the beginning of the entire string.

Reply:
or does that cross the line length limit?

Reply:
Updated